salt/_modules/caasp_hosts.py

```diff
@@ -68,13 +68,25 @@ def __virtual__():
     return "caasp_hosts"
 
 
-def _concat(lst1, lst2):
+# returns a list resulting of appending `lst2` to `lst1`, removing duplicates on
+# both lists (not preserving order on any of them) and removing empty elements on
+# the result. The result will be sorted as well
+def _sorted_append(lst1, lst2):
     res = list(set(lst1) | set(lst2))  # join both lists (without dups)
     res = [x for x in res if x]  # remove empty strings
     res.sort()  # sort the result (for determinism)
     return res
 
 
+# returns a list resulting of prepending `lst2` to `lst1`, not removing
+# duplicates on any of both lists (preserving order on both of them) and
+# removing empty elements on the result
+def _unsorted_prepend(lst1, lst2):
+    res = lst2 + lst1  # unsorted prepend of lst2 in lst1
+    res = [x for x in res if x]  # remove empty strings
+    return res
```

Comment:
Yes, we could even reduce the whole function to `return [x for x in lst2 + lst1 if x]`, but I think we are leaving it as it is because it makes the comments more clear...

Reply:
yop, i think 1 liner is to complicate but imho 2 lines could be ok  but is ok also 3

Reply:
Leaving to keep the style with the previous code, refactoring more things here is out of the scope for this PR.